packages/server/api/controllers/users.controller.js

```diff
@@ -0,0 +1,36 @@
+const knex = require('../../config/db');
+const HttpError = require('../lib/utils/http-error');
+
+// get users
+const getUsers = async () => {
+  try {
+    const users = await knex.select().table('users');
+    if (users.length === 0) {
+      throw new HttpError('No users found', 404);
+    }
+    return users;
+  } catch (error) {
+    return error.message;
+  }
+};
+
+// post
+const createUsers = async (body) => {
+  try {
+    await knex('users').insert({
+      full_name: body.full_name,
+      email: body.email,
+      uid: body.uid,
```

Comment:
Did you figure out a way to deal with duplicate uids? Couldn't that happen if someone tries to sign up twice with the same email?

Reply:
it goes through firebase first, so user is not created if someone tries to sign up twice with the same email;